PlayCover/Utils/LegacySettings.swift

```diff
@@ -0,0 +1,158 @@
+//
+//  LegacySettings.swift
+//  PlayCover
+//
+//  Created by Isaac Marovitz on 07/09/2022.
+//
+
+// swiftlint:disable function_body_length
+
+import Foundation
+
+class LegacySettings {
+    public static var monolithURL: URL = URL(fileURLWithPath:
+                                                "/Users/\(NSUserName())/Library/Preferences/playcover.plist")
+    public static var doesMonolithExist: Bool {
+        return FileManager.default.fileExists(atPath:
+                                                "/Users/\(NSUserName())/Library/Preferences/playcover.plist")
+    }
+
+    static func convertLegacyMonolithPlist(_ from: URL) {
+        var dictionary: [String: Any]
+
+        do {
+            if let monolith = try [String: Any].read(from) {
+                if !monolith.isEmpty {
+                    dictionary = monolith
+
+                    for (key, value) in dictionary {
+                        if let dict = value as? [String: Any] {
+                            if let settings = convertLegacySettingsDict(dict) {
+                                let settingsURL = AppSettings.appSettingsDir.appendingPathComponent("\(key).plist")
+                                do {
+                                    let data = try PropertyListEncoder().encode(settings)
+                                    try data.write(to: settingsURL)
+                                } catch {
+                                    print(error)
+                                }
+                            }
+
+                            if let legacyKeymaps = dict["pc.layout"] as? [Any] {
+                                let keymap = convertLegacyKeymapArray(legacyKeymaps, key)
+                                let keymapURL = Keymapping.keymappingDir.appendingPathComponent("\(key).plist")
+                                do {
+                                    let data = try PropertyListEncoder().encode(keymap)
+                                    try data.write(to: keymapURL)
+                                } catch {
+                                    print(error)
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        } catch {
+            Log.shared.error("Failed to read playcover.plist")
+        }
+    }
+
+    static func convertLegacySettingsDict(_ from: [String: Any]) -> AppSettingsData? {
+        var dictionary: [String: Any]
+
+        if !from.isEmpty {
+            dictionary = from
+
+            var settingsData = AppSettingsData()
```

Comment:
The legacy keymap settings also has:
```suggest
pc.bypass, pc.enableWindowAutoSize, pc.gameWindowSizeHeight, pc.gameWindowSizeWidth, pc.ipadModel
```

Reply:
Those are not keymapping, those are settings that can be changed in menu of they're still available.

Reply:
This PR is aiming to preserve settings too.

Reply:
Here's my rational for excluding those parameters.

`pc.bypass`: Jailbreak Bypass did not work as intended in the majority of legacy PlayCover
`pc.enableWindowAutoSize`, `pc.gameWindowSizeHeight`, `pc.gameWindowSizeWidth`: The way that resolution behaves in 2.0 has been almost entirely refactored. Setting new settings based on this data might cause unintended behaviour.
`pc.ipadModel`: M1 Model Identifier <2.0 was broken, now that has been fixed, the default has been changed to M1